src/main/java/org/folio/inventory/resources/MoveApi.java

```diff
@@ -103,41 +117,97 @@ private void updateItems(RoutingContext routingContext, WebContext context, List
-        List<String> nonUpdatedIds = ListUtils.subtract(itemIdsToUpdate, updatedItemIds);
-        HttpServerResponse response = routingContext.response();
-        if (nonUpdatedIds.isEmpty()) {
-          JsonResponse.successWithEmptyBody(response);
-        } else {
-          JsonResponse.successWithIds(response, nonUpdatedIds);
-        }
-      });
+      .thenAccept(updatedIds -> respond(routingContext, idsToUpdate, updatedIds));
   }
 
-  private OkapiHttpClient createHttpClient(RoutingContext routingContext, WebContext context) throws MalformedURLException {
-    return new OkapiHttpClient(client, context, exception -> ServerErrorResponse.internalError(routingContext.response(),
-        String.format("Failed to contact storage module: %s", exception.toString())));
+  private void moveHoldings(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject holdingsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = holdingsMoveHasRequiredFields(holdingsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      String toInstanceId = holdingsMoveJsonRequest.getString(TO_INSTANCE_ID);
+      List<String> holdingsRecordsIdsToUpdate = toListOfStrings(holdingsMoveJsonRequest.getJsonArray(HOLDINGS_RECORD_IDS));
+      storage.getInstanceCollection(context)
+        .findById(toInstanceId)
+        .thenAccept(instance -> {
+          if (Objects.nonNull(instance)) {
+            try {
+              OkapiHttpClient okapiClient = createHttpClient(routingContext, context);
+              CollectionResourceClient holdingsStorageClient = createStorageClient(okapiClient, context, HOLDINGS_STORAGE);
+              MultipleRecordsFetchClient multipleRecordsFetchClient = createFetchClient(holdingsStorageClient,
+                  HOLDINGS_RECORDS_PROPERTY);
+
+              multipleRecordsFetchClient.find(holdingsRecordsIdsToUpdate, this::fetchByIdCql)
+                .thenAccept(jsons -> {
+                  List<HoldingsRecord> holdingsRecordsToUpdate = jsons.stream()
+                    .map(json -> json.mapTo(HoldingsRecord.class))
+                    .map(holding -> holding.withInstanceId(toInstanceId))
+                    .collect(toList());
+                  updateHoldings(routingContext, context, holdingsRecordsIdsToUpdate, holdingsRecordsToUpdate);
+                })
+                .exceptionally(e -> {
+                  ServerErrorResponse.internalError(routingContext.response(), e);
+                  return null;
+                });
+            } catch (Exception e) {
+              ServerErrorResponse.internalError(routingContext.response(), e);
+            }
+          } else {
+            JsonResponse.unprocessableEntity(routingContext.response(), "Instance with id=" + toInstanceId + " not found");
+          }
+        });
```

Comment:
Can it be refactored and some logical part extracted to separate methods, please?